### PR TITLE
Add gsw infunnel

### PR DESCRIPTION
**Pull request by ocefpaf**
This PR adds the infunnel function ported from the Matlab version.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@efiring I tested using some random values and comparing against the Matlab version. Seems to work but I'm not a c programmer, in fact I never tried anything in  C until this function. I believe the actual function is OK but the tests are definitely kind of strange when compared with the rest of the tests in there.

I'd love to get this into both R and Python wrappers though. Let me know what you think.

File: gsw_check_functions.c [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #include "gswteos-10.h"
 #include "gsw_check_data.h"
 
+#include <stdio.h>
```
Comment (efiring) [inline gsw_check_functions.c:11]: It's already included above, so you can delete this line.

File: gsw_check_functions.c [hunk 2]
```diff
@@ -96,6 +98,32 @@ double          tf_poly[cast_m*cast_n];
...
+    } else {
+        printf("%s: Failed (Expected: %d, Actual: %d)\n", test_name, expected, actual);
+        return gsw_error_flag=1;
+    }
+}
+
+
+void test_infunnel() {
+    // Inside the funnel.
+    int result1 = gsw_infunnel(35.0, 5.0, 400);
```
Comment (efiring) [inline gsw_check_functions.c:113]: I would put a single declaration "int result;" after the opening brace, and then use that in place of the numbered versions.  Nothing is gained by using different variables.  Although the most modern C allows variable declarations in other locations, most C still follows the earlier pattern of keeping variable declarations restricted to being after an opening brace (of the function, or of a scope within the function).

File: gsw_oceanographic_toolbox.c [hunk 3]
```diff
@@ -180,6 +180,39 @@ gsw_add_mean(double *data_in, double *data_out)
...
+*/
+int
+gsw_infunnel(double sa, double ct, double p)
+{
+    return !(p > 8000 ||
+        sa < 0 ||
+        sa > 42 ||
+        (p < 500 && ct < gsw_ct_freezing(sa, p, 0)) ||
+        (p >= 500 && p < 6500 && sa < p * 5e-3 - 2.5) ||
+        (p > 500 && p < 6500 && ct > (31.66666666666667 - p * 3.333333333333334e-3)) ||
```
Comment (efiring) [inline gsw_oceanographic_toolbox.c:208]: This looks like a trivial bug in the Matlab version.  From the pattern here, which matches the Matlab, it is obvious that the first condition should be "p >= 500", not "p > 500".

**Review by efiring — CHANGES_REQUESTED** [review]
Minor changes.

File: gsw_oceanographic_toolbox.c (hunk 3 above)
Comment (ocefpaf) [reply gsw_oceanographic_toolbox.c]: Fixed!

File: gsw_check_functions.c (hunk 2 above)
Comment (ocefpaf) [reply gsw_check_functions.c]: Thanks! Just learned something new. Learning C as I go here 😬

**Comment by ocefpaf** [comment]
> Minor changes.

@efiring let me know if there is anything else we need to do here.

**Review by efiring — APPROVED** [review]
Merge when ready.

**Pull request merged**